### PR TITLE
Update gen-pack.sh to be platform agnostic by using

### PR DESCRIPTION
**Pull request by JonatanAntoni**
- UTILITY_GIT and UTILITY_GHCLI instead of hard-coded tools
- unarchive function instead of hard-coded 7z

### CODE REVIEW

File: gen_pack.sh [hunk 1]
```diff
@@ -87,28 +87,29 @@ PACKCHK_DEPS="
 # usage: preprocess <build>
 #   <build>  The build folder
 #
+# shellcheck disable=SC2317
```
Comment (RobertRostohar) [inline gen_pack.sh:90]: Why is this needed?

Comment (JonatanAntoni) [reply gen_pack.sh:90]: I have shellcheck installed in my Code and it complains about the function beging "unused" because it can't detect its usage.

**Review by RobertRostohar — APPROVED** [review]
